modules/outbound-command-event-handler/src/domain/bulk_transaction_agg/index.ts

```diff
@@ -156,6 +178,130 @@ export class BulkTransactionAgg extends BaseAggregate<BulkTransactionEntity, Bul
+        await (<IBulkTransactionEntityRepo> this._entity_state_repo)
+            .setBulkQuotesFailedCount(this._rootEntity.id, count);
+    }
+    async incrementBulkQuotesFailedCount() : Promise<void> {
+        await (<IBulkTransactionEntityRepo> this._entity_state_repo)
+            .incrementBulkQuotesFailedCount(this._rootEntity.id);
+    }
+
+    async addBulkBatchEntity(entity: BulkBatchEntity) : Promise<void> {
+        await (<IBulkTransactionEntityRepo> this._entity_state_repo)
+            .setBulkBatch(this._rootEntity.id, entity.id, entity.exportState());
+    }
+
+    async setBulkBatchById(id: string, bulkBatch: BulkBatchEntity): Promise<void> {
+        await (<IBulkTransactionEntityRepo> this._entity_state_repo)
+            .setBulkBatch(this._rootEntity.id, id, bulkBatch.exportState());
+    }
+
+    async createBatches(maxItemsPerBatch: number) : Promise<void> {
+        const allBulkBatchIds = await this.getAllBulkBatchIds()
+        if (allBulkBatchIds.length > 0) {
+            throw(new Error('Bulk batches are already created on this aggregator'));
+        }
+
+        const batchesPerFsp: {[fspId: string]: string[][]} = {}
+        // Iterate through individual transfers
+        const allIndividualTransferIds = await this.getAllIndividualTransferIds();
+        for await (const individualTransferId of allIndividualTransferIds) {
+            // Create the array of batches per each DFSP with maximum limit from the config containing Ids of individual transfers
+            const individualTransfer = await this.getIndividualTransferById(individualTransferId);
+            if (individualTransfer.transferState === 'DISCOVERY_ACCEPTED' && individualTransfer.toFspId) {
+                // If there is any element with fspId
+                if(batchesPerFsp[individualTransfer.toFspId]) {
+                    const batchFspIdArray = batchesPerFsp[individualTransfer.toFspId];
+                    const batchFspLength = batchFspIdArray.length;
+                    const lastElement = batchFspIdArray[batchFspLength - 1];
+                    // If the length reaches maximum value, create new element and insert the id
+                    if (lastElement.length < maxItemsPerBatch) {
+                        lastElement.push(individualTransfer.id)
+                    } else {
+                        const newElement = [ individualTransfer.id ];
+                        batchFspIdArray.push(newElement)
+                    }
+                } else {
+                    const newElement = [ individualTransfer.id ];
+                    batchesPerFsp[individualTransfer.toFspId] = [];
+                    batchesPerFsp[individualTransfer.toFspId].push(newElement);
+                }
+            } else {
+                this._logger.error(`The individual transfer with id ${individualTransfer.id} is not in state DISCOVERY_ACCEPTED or toFspId is not found in the partyResponse`)
```

Comment:
Is this an error, or rather a warning?

If its an error, then shouldn't we be throwing one?

Reply:
This is an error and we are ignoring that particular transfer encase of this error. We shouldn't throw error to continue creating batches. May be we should update the lastError param on that individual entity but there should be some structure define for that parameter before that.